homeassistant/components/hyperion/light.py

```diff
@@ -223,83 +226,51 @@ async def async_setup_entry(
     hass: HomeAssistantType, config_entry: ConfigEntry, async_add_entities: Callable
 ) -> bool:
     """Set up a Hyperion platform from config entry."""
-    host = config_entry.data[CONF_HOST]
-    port = config_entry.data[CONF_PORT]
-    token = config_entry.data.get(CONF_TOKEN)
 
-    async def async_instances_to_entities(response: Dict[str, Any]) -> None:
-        if not response or const.KEY_DATA not in response:
-            return
-        await async_instances_to_entities_raw(response[const.KEY_DATA])
-
-    async def async_instances_to_entities_raw(instances: List[Dict[str, Any]]) -> None:
-        registry = await async_get_registry(hass)
-        entities_to_add: List[HyperionLight] = []
-        desired_unique_ids: Set[str] = set()
-        server_id = cast(str, config_entry.unique_id)
-
-        # In practice, an instance can be in 3 states as seen by this function:
-        #
-        #    * Exists, and is running: Add it to hass.
-        #    * Exists, but is not running: Cannot add yet, but should not delete it either.
-        #      It will show up as "unavailable".
-        #    * No longer exists: Delete it from hass.
-
-        # Add instances that are missing.
-        for instance in instances:
-            instance_id = instance.get(const.KEY_INSTANCE)
-            if instance_id is None or not instance.get(const.KEY_RUNNING, False):
-                continue
-            unique_id = get_hyperion_unique_id(
-                server_id, instance_id, TYPE_HYPERION_LIGHT
-            )
-            desired_unique_ids.add(unique_id)
-            if unique_id in current_entities:
-                continue
-            hyperion_client = await async_create_connect_hyperion_client(
-                host, port, instance=instance_id, token=token
-            )
-            if not hyperion_client:
-                continue
-            current_entities.add(unique_id)
-            entities_to_add.append(
+    entry_data = hass.data[DOMAIN][config_entry.entry_id]
+    server_id = config_entry.unique_id
+
+    def instance_add(instance_num: int, instance_name: str) -> None:
```

Comment:
This needs to be decorated with `@callback` (imported from `homeassistant.core`) or otherwise the dispatcher will execute this method in the executor.

```suggestion
    @callback
    def instance_add(instance_num: int, instance_name: str) -> None:
```

This needs to be done for all methods that are passed to the dispatcher, but generally it's good practice to do this to all methods that are to be run inside the event loop. It makes it easier for readers of the code to see what is going on .

Reply:
Got it. Thank you! This page helped: https://developers.home-assistant.io/docs/asyncio_categorizing_functions/#the-callback-function